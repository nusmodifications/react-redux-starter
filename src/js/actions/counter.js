// @flow
import type { FSA } from 'types/redux';

export const INCREMENT = 'INCREMENT';
export function increment(): FSA {
  return {
    type: INCREMENT,
    payload: null,
  };
}

export const DECREMENT = 'DECREMENT';
export function decrement(): FSA {
  return {
    type: DECREMENT,
    payload: null,
  };
}
