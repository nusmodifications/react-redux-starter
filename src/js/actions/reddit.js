// @flow

import { API_REQUEST } from 'middlewares/requests-middleware';

export const FETCH_REDDITS: string = 'FETCH_REDDITS';
export function fetchReddits(topic: string): Function {
  return (dispatch: Function) => dispatch({
    [API_REQUEST]: {
      type: FETCH_REDDITS,
      payload: {
        method: 'GET',
        url: `https://www.reddit.com/r/${topic}.json`,
      },
    },
  });
}
