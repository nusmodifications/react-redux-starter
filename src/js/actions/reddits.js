// @flow
import type { FSA } from 'types/redux';
import { API_REQUEST } from 'middlewares/requests-middleware';

export const FETCH_REDDITS: string = 'FETCH_REDDITS';
export function fetchReddits(topic: string): FSA {
  return {
    type: FETCH_REDDITS,
    payload: {
      method: 'GET',
      url: `https://www.reddit.com/r/${topic}.json`,
    },
    meta: {
      [API_REQUEST]: true,
    },
  };
}
