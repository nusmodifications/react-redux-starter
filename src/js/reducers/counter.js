// @flow
import type { FSA } from 'redux';

import { INCREMENT, DECREMENT } from 'actions/counter';

export type CounterState = number;
export const defaultCounterState: CounterState = 0;

function counter(state: CounterState = defaultCounterState, action: FSA): CounterState {
  switch (action.type) {
    case INCREMENT:
      return state + 1;
    case DECREMENT:
      return state - 1;
    default:
      return state;
  }
}

export default counter;
