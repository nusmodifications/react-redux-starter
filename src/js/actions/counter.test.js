import type { FSA } from 'types/redux';

import * as actions from 'actions/counter';

describe('counter actions', () => {
  it('should create an action to increment', () => {
    const action: FSA = actions.increment();
    expect(action).toMatchSnapshot();
  });

  it('should create an action to decrement', () => {
    const action: FSA = actions.decrement();
    expect(action).toMatchSnapshot();
  });
});
