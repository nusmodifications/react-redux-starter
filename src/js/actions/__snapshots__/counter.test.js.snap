// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter actions should create an action to decrement 1`] = `
Object {
  "payload": null,
  "type": "DECREMENT",
}
`;

exports[`counter actions should create an action to increment 1`] = `
Object {
  "payload": null,
  "type": "INCREMENT",
}
`;
